### ISSUE TITLE
sizing issue with custom style

### ISSUE THREAD

**Issue by morganwu** [PolymerElements/paper-dialog#54]
I'm trying to define a custom pop up window to make it responsive for different viewports, based on paper-dialog-behavior. I want it to take full size of screen in mobile screen, just like a new full screen view in native apps to be pushed into, and have a limited size in pc screen. I used paper-dialog for testing, but found the styles are not applied correctly and the dialog is not kept centered when the window is being resized. Maybe I misused styles, please enlighten me if anyone have any idea. Following is my test code, I described the expecting behaviours and the testing results inside. Thanks.

``` html
<!doctype html>
<html lang="">
<head>
    <meta charset="utf-8">
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <meta name="mobile-web-app-capable" content="yes">
    <script src="../bower_components/webcomponentsjs/webcomponents-lite.js"></script>
    <link rel="import" href="../bower_components/paper-dialog/paper-dialog.html">
    <link rel="import" href="../bower_components/paper-button/paper-button.html">
    <link rel="import" href="../bower_components/iron-icons/iron-icons.html">
    <link rel="import" href="../bower_components/iron-icon/iron-icon.html">
    <style is="custom-style">
        paper-dialog {
            background: var(--paper-indigo-a200);
            color: #FFFFFF;
        }

        @media (max-width: 600px) {
            paper-dialog {
                position: absolute;
                top: 0;
                right: 0;
                bottom: 0;
                left: 0;
                margin: 0;
            }
        }

        @media (min-width: 601px) {
            paper-dialog {
                width: calc(100% - 20px);
                max-width: 800px;
                height: calc(100% - 20px);
                max-height: 800px;
            }
        }

        iron-icon {
            display: inline-block;
        }

        ul {
            list-style-type: none;
            padding-left: 0;
        }

        li {
            margin-left: 0;
        }
    </style>
</head>

<body unresolved class="fullbleed layout vertical">
<span id="browser-sync-binding"></span>
<paper-button style="position: fixed; right: 40px; bottom: 40px;" onclick="clickHandler()">open dialog</paper-button>
<paper-dialog id="dialog" entry-animation="scale-up-animation" exit-animation="fade-out-animation">
    <section>
        <!--<h4>Please try to resize window from pc size and mobile size, and check the results.</h4>-->
        <span>Expecting behaviors and test results:</span>

        <h4>Dialog take the full size of screen in mobile mode (max-width <= 600px)</h4>
        <ul>
            <li>
                <iron-icon icon="icons:check"></iron-icon>
                Case 1, start from mobile size.
            </li>
            <li>
                <iron-icon icon="icons:clear"></iron-icon>
                Case 2, start from pc size, resize to mobile size.
            </li>
        </ul>

        <h4>Dialog will be centered all time window is being resized, respecting max-height and max-width style</h4>
        <ul>
            <li>
                <iron-icon icon="icons:clear"></iron-icon>
                Case 1, start from pc size, resize to mobile size, and back to pc size.
            </li>
            <li>
                <iron-icon icon="icons:clear"></iron-icon>
                Case 2, start from mobile size, resize to pc size.
            </li>
            <li>
                <iron-icon icon="icons:clear"></iron-icon>
                Case 3, open dialog, dismiss it, and open again.
            </li>
        </ul>

    </section>
</paper-dialog>

<script type="application/javascript">
    function clickHandler() {
        document.querySelector('paper-dialog').open();
    }
</script>
</body>

</html>

```

**Comment by valdrinkoshi** [comment]
hi @MorganWu,

tl;dr this does the trick:

``` html
<style is="custom-style">
  paper-dialog {
    background: var(--paper-indigo-a200);
    color: #FFFFFF;
    width: 100%;
    height: 100%;
    max-width: 800px;
    max-height: 800px;
  }

  @media (max-width: 600px) {
    paper-dialog {
      margin: 0;
    }
  }

  iron-icon {
    display: inline-block;
  }

  ul {
    list-style-type: none;
    padding-left: 0;
  }

  li {
    margin-left: 0;
  }
</style>
```

You don't need to set `position: absolute`, and setting `margin: auto` will center the dialog on the page. Also, no need to have 2 @media selectors, 1 is enough.

**Comment by valdrinkoshi** [comment]
Actually, this still gives issues. I noticed that when resizing the window frame, the dialog doesn't get positioned properly. Also, it gets rendered correctly at the first open, but wrongly in the following ones.
I suspect this is a bug on `IronFitBehavior`. Re-opening.

**Comment by valdrinkoshi** [comment]
So, this is caused by https://github.com/PolymerElements/iron-fit-behavior/issues/29

**Comment by morganwu** [comment]
hi @valdrinkoshi ,

It works perfectly, exactly what I want. Thank you.

**Comment by valdrinkoshi** [comment]
awesome! closing this.
